packages/css-data/src/property-parsers/linear-gradient.ts

```diff
@@ -0,0 +1,213 @@
+            const value = gradientParts
+              .map((item) => csstree.generate(item))
+              .join(" ");
+            sideOrCorner = { type: "keyword", value };
+          }
+
+          // if there is a color-stop in the gradientParts, then we need to parse it for position and hint.
+          const colorStop = gradientParts.find(isColorStop);
+          if (colorStop !== undefined) {
+            const [_, position, hint] = gradientParts;
+
+            const stop: GradientStop = {
+              color: getColor(colorStop),
+              position:
+                position &&
+                (position.type === "Percentage" ||
+                  position.type === "Dimension")
+                  ? {
+                      type: "unit",
+                      value: parseFloat(position.value),
+                      unit:
+                        position.type === "Percentage"
+                          ? "%"
+                          : (position.unit as Unit),
+                    }
+                  : undefined,
+              hint:
+                hint &&
+                (hint.type === "Percentage" || hint.type === "Dimension")
+                  ? {
+                      type: "unit",
+                      value: parseFloat(hint.value),
+                      unit:
+                        hint.type === "Percentage" ? "%" : (hint.unit as Unit),
+                    }
+                  : undefined,
+            };
+
+            stops.push(stop);
+          }
+
+          gradientParts = [];
+        }
+      }
+    }
+  });
+
+  return { angle, sideOrCorner, stops };
+};
+
+const isAngle = (node: csstree.CssNode): node is csstree.Dimension =>
+  node.type === "Dimension" &&
+  ["deg", "grad", "rad", "turn"].includes(node.unit);
+
+const isSideOrCorner = (node: csstree.CssNode): node is csstree.Identifier =>
+  node.type === "Identifier" && sideOrCorderIdentifiers.includes(node.name);
+
+const isColorStop = (
+  node: csstree.CssNode
+): node is csstree.Identifier | csstree.FunctionNode | csstree.Hash => {
+  return (
+    (node.type === "Function" ||
+      (node.type === "Identifier" &&
+        sideOrCorderIdentifiers.includes(node.name)) === false ||
+      node.type === "Hash") &&
+    colord(csstree.generate(node)).isValid() === true
+  );
+};
+
+const getColor = (
+  node: csstree.FunctionNode | csstree.Identifier | csstree.Hash
+): RgbValue | undefined => {
+  let color: string;
+  if (node.type === "Function") {
+    color = csstree.generate(node);
+  } else if (node.type === "Identifier") {
+    color = node.name;
+  } else {
+    color = `#${node.value}`;
+  }
+
+  const result = colord(color);
+  if (result.isValid()) {
+    const value = result.toRgb();
+
+    return {
+      type: "rgb",
+      r: value.r,
+      g: value.g,
+      b: value.b,
+      alpha: value.a,
+    };
+  }
+};
+
+export const reconstructLinearGradient = (parsed: ParsedGradient): string => {
```

Comment:
Why not just use toValue with original StyleValue?

Reply:
I think it's best to save gradients as `unparsed` in the data-layer. Because they already come under as a `tuple` for background.
And secondary we parse the `gradient` just for showing it in the component. So, it's best to read and save it back as `unparsed` itself. And to `reconstruct` is just to save it back as `unparsed to the dat layer. 

The actual implementation can be differ when we start building the feature.  Right now, i feel keeping it as `ParsedGradient` for the component works best to keep the `stops` and `hints` and `angle` at disposal.